### DISCUSSION TITLE
Not able to add logical model using big query as data source

### DISCUSSION THREAD

**Discussion by mukeshkrgupta1** [hasura/graphql-engine#9733] [Question] [answered]
@SamirTalwar I was trying to explore group by. I am using big query as connection. Using v2.28.0-cloud.1 version. As per document, native query is supported from version v2.26.0.  Getting error during creation of add logical model. Attached error screenshot
![logical-data-model-error-hasura-cloud](https://github.com/hasura/graphql-engine/assets/118717885/fa55fc93-e821-47a9-9a7d-539db2c7d974)
ot

**Comment by SamirTalwar** [comment]
@mukeshkrgupta1, can you please share a little more information?

Specifically, I would really like to see the request data. Could you make the same request again, as follows:

1. Open the _Developer Tools_ in your browser.
2. Move to the _Network_ tab.
3. Click the "Create Logical Model" button to make the request.
4. Find the network event corresponding to the metadata change.
5. Copy the contents of the _Request_ and _Response_ tabs into a comment here (redacting any sensitive information). You may need to ask for the "raw" JSON data to copy it.

Please also make sure that the version of the Console you are using is the same as the version on the server. If you are starting the Console from the Hasura CLI, you may need to update it first.

**Reply by mukeshkrgupta1** [reply]
@SamirTalwar I am using hasura cloud version: v2.28.0-cloud.1.  I am not starting the Console from the Hasura CLI. Just used hasura cloud and connected with big query as data source.

**Note:** As per instruction demo on [youtube](https://www.youtube.com/watch?v=1EMq4RVB2XA&t=893s)/[article](https://hasura.io/docs/latest/schema/bigquery/logical-models/native-queries/) I skipped step: Create table and insert table step. As big query table is already created. Also I am able to access GraphQL Endpoint such as https://big-query-demo.hasura.app/v1/graphql. Also able to run SQL query on big query table using hasura cloud after connection with big query data set. Want to explore groupBy so trying native query here. Also please let us know what is roadmap to support groupBy without native query?


**Please find request:**

```json
{
  "resource_version": 7,
  "type": "bigquery_track_logical_model",
  "args": {
    "source": "xx-production_analytics",
    "name": "cdc_sold_products_model",
    "fields": [
      {
        "name": "id",
        "type": "text",
        "nullable": true
      },
      {
        "name": "timestamp",
        "type": "TIMESTAMP",
        "nullable": true
      },
      {
        "name": "price_amount",
        "type": "FLOAT64",
        "nullable": true
      }
..........
    ]
  }
}
```

**Response**

```json
{
    "error": "Error when parsing command track_logical_model.\nSee our documentation at https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/index.html#metadata-apis.\nInternal error message: \n  Previous branch failure: Error in $.type: Error in $: parsing LogicalModelTypeReference failed, expected Object, but encountered String\nError in $: Did not recognize scalar type 'text'",
    "path": "$.args.fields[0].type",
    "code": "parse-failed"
}
```

**Reply by mukeshkrgupta1** [reply]
@SamirTalwar could you help here. Let me know if any details required.

**Reply by vijayprasanna13** [reply]
@mukeshkrgupta1 we're able to replicate this issue on our side as well and it seems to be a console display issue. (if you manually pick "STRING" from the list of types for the field type, it actually works right now). 
I'll update the thread once we have a fix!

**Reply by vijayprasanna13** [reply, marked as answer]
@mukeshkrgupta1 this should be fixed now on `v2.28.0-cloud.1`. Could you try it out again?

**Reply by mukeshkrgupta1** [reply]
Working fine. Thanks.
